bigchaindb/backend/mongodb/connection.py

```diff
@@ -113,6 +113,8 @@ def _connect(self):
                                              replicaset=self.replicaset,
                                              serverselectiontimeoutms=self.connection_timeout,
                                              ssl=self.ssl)
+                if self.login is not None and self.password is not None:
```

Comment:
Only `login` is required for X509, and both `login` and `password` are required for SCRAM.